lib/git-manager.js

```diff
@@ -373,66 +342,43 @@ function parseRepo(repo) {
 /**
  * Return a promise for all issue events for the given issue / pull
  */
-function getIssueEvents(repo, issueNumber) {
-   debug("Getting events for issue #%s", issueNumber);
-   return api.getIssueEvents(params({
-      number: issueNumber
-   }, repo)).then(getAllPages);
+function getIssueEvents(repo, number) {
+   debug("Getting events for issue #%s", number);
+   return github.issues.getEvents(params({ number }, repo))
+                       .then(paginate);
 }
 
-function getIssueComments(repo, issueNumber) {
-   debug("Getting comments for issue #%s", issueNumber);
-   return api.getIssueComments(params({
-      number: issueNumber
-   }, repo)).then(getAllPages);
+function getIssueComments(repo, number) {
+   debug("Getting comments for issue #%s", number);
+   return github.issues.getComments(params({ number }, repo))
+                       .then(paginate);
 }
 
-function getPullReviewComments(repo, pullNumber) {
-   debug("Getting pull review comments for pull #%s", pullNumber);
-   return api.getPullReviewComments(params({
-      number: pullNumber
-   }, repo)).then(getAllPages);
+function getPullReviewComments(repo, number) {
+   debug("Getting pull review comments for pull #%s", number);
+   return github.pullRequests.getComments(params({ number }, repo))
+                .then(paginate);
 }
 
 function getCommit(repo, sha) {
-   return new Promise(function (resolve, reject){
-      debug("Getting commit %s", sha);
-      api.getCommit(params({
-         sha: sha,
-         per_page: 1
-      }, repo)).then(function(commits) {
-         if (commits.length === 1) {
-            resolve(commits[0]);
-         } else {
-            return reject();
-         }
-      });
-   });
+   return github.repos.getCommit(params({ sha }, repo))
+                      .then(res => res.data);
```

Comment:
We only use `getCommit` in `module.exports.parse` and we don't have any promise catch handling on the response from this function, so I don't think it really has an effect. The shape of the response is now a single object instead of a list too, so the error case doesn't seem relevant anymore.